process/generators.py

```diff
@@ -30,12 +38,14 @@ def gen_units(proj):
 
     # TODO: how to handle cases where better numbers exist from dbi
```

Comment:
Just for my own understanding, does this mean that with this new logic of passign in source then at places like here where we need to still do the 'weighting' system the caller will just have to carry out that logic?

Reply:
yeah. honestly the graph-traversing logic was my own brainfart from compartmentalizing some early hacky code and not changing it to this generator system, where the business logic really should live in terms of how to determine whether to use e.g. pts vs ppts.

Reply:
Ahh you're right, long as we're making sure to keep track of that inconsistency